web/base_viewer.js

```diff
@@ -612,6 +626,11 @@ class BaseViewer {
               }
             );
           }
+
+          if (isPureXfa) {
+            return Promise.all(xfaDrawPromises);
```

Comment:
The return value of this function isn't used for anything, hence I don't think this actually does what you think here. Basically, I'm assuming that you'd want to delay `this._pagesCapability.resolve();` in the XFA-case (however please note the comment regarding the overall implementation.)